### PR TITLE
feat: Add BeforeSendTransaction hook

### LINKED ISSUE #500: Add beforeSendTransaction
Problem Statement
Similar to beforeSend, we will add beforeSendTransaction to allow people to strip Pii or drop transactions before they are sent to Sentry.

Solution Brainstorm
https://github.com/getsentry/sentry-javascript/pull/6121

### PR DESCRIPTION
**Pull request by tonyo**
`BeforeSendTransaction` works similarly to `BeforeSend`, but it is applied only to transaction events.

Closes #500.

Docs:
* [ ] Basic Options page: https://github.com/getsentry/sentry-docs/pull/6039
* [ ] Filtering page: https://github.com/getsentry/sentry-docs/pull/6049

### CODE REVIEW

**Review by cleptric — COMMENTED** [review]
Is there already a docs PR for this?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+### Features
+
+* Add `beforeSendTransaction` hook ([#517](https://github.com/getsentry/sentry-go/pull/517))
```
[the remark this answers is not in the document]
Comment (tonyo) [reply CHANGELOG.md:7]: I guess adding a link that point to the docs (when they're ready) should be enough?
This change doesn't seem to be too important or breaking enough to cover it in the changelog.

[the remark this answers is not in the document]
Comment (cleptric) [reply CHANGELOG.md:7]: I would like our changelogs to be a bit more telling, especially when it comes to new features. The Ruby SDK is a good benchmark we should aim for https://github.com/getsentry/sentry-ruby/releases

**Comment by tonyo** [comment]
Added an example and updated the changelog.
Docs are pending for now, as described in https://github.com/getsentry/sentry-docs/pull/6039

**Review by cleptric — APPROVED** [review]

**Pull request merged**